blobfuse/blobfuse.cpp

```diff
@@ -327,28 +341,45 @@ void *azs_init(struct fuse_conn_info * conn)
     conn->max_background = 128;
     //  conn->want |= FUSE_CAP_WRITEBACK_CACHE | FUSE_CAP_EXPORT_SUPPORT; // TODO: Investigate putting this back in when we downgrade to fuse 2.9
 
-
-    if (libcurl_version < 7.54) {
-	// Curl was not intialized for lower version as that results into
-	// failure post fork. So tls and cpplite were not initialized pre-fork for lower
-	// kernel version. Do the init now before starting.
-	syslog(LOG_CRIT, "** Post fork authentication for older libcurl version");
-
-	configure_tls();
-	if(storage_client->AuthenticateStorage())
-	{
-	    syslog(LOG_DEBUG, "Successfully Authenticated!");   
-	}
-	else
-	{
-	    fprintf(stderr, "Unable to start blobfuse due to a lack of credentials. Please check the readme for valid auth setups.");
-	    syslog(LOG_ERR, "Unable to start blobfuse due to a lack of credentials. Please check the readme for valid auth setups.");
-            exit(1);
-	}
-    }
-
     g_gc_cache = std::make_shared<gc_cache>(config_options.tmpPath, config_options.fileCacheTimeoutInSeconds);
     g_gc_cache->run();
+    
+    if (libcurl_version < blobfuse_constants::minCurlVersion) {
+        // Curl was not intialized for lower version as that results into
+        // failure post fork. So tls and cpplite were not initialized pre-fork for lower
+        // kernel version. Do the init now before starting.
+        syslog(LOG_CRIT, "** Post fork authentication for older libcurl version");
+
+        configure_tls();
+        if(storage_client->AuthenticateStorage())
+        {
+            syslog(LOG_DEBUG, "Successfully Authenticated!");   
+        }
+        else
+        {
+            char errStr[] = "Unable to start blobfuse due to a lack of credentials. Please check the readme for valid auth setups.\n";
+            if (stdErrFD != -1) {
+                ssize_t n = write(stdErrFD, errStr, sizeof(errStr));
+                if (n == -1) {
+                    syslog(LOG_ERR, "Failed to report back the status of auth.");
+                }
+            }
+            syslog(LOG_ERR, "%s", errStr);
+            std::thread t1(std::bind(&destroyBlobfuseOnAuthError));
+            t1.detach();
```

Comment:
does this detach succeed even if the main thread is executing? In my testing if I am in that dir it did not unmount.
Can we throw an error if unmount fails so that the user knows they have to unmount manually?

Reply:
fuse_unmount is of void return type so we need to explicitly check whether unmount was successful or not. By the call itself we can not judge it.

Reply:
Have added mount directory check post unmount for the error reporting.